apps/guardian/src-tauri/src/main.rs

```diff
@@ -1,130 +1,426 @@
+            account: wallet_account.address,
+            keys: wallet_account.keys,
+            index,
+        }
+    }
+}
+
+/// Holds the currently selected account and corresponding password
+struct ActiveGuardian {
+    /// The guardian data for the active account
+    guardian: GuardianData,
     /// The password used for encryption with the selected account
-    #[allow(dead_code)] // TODO: remove when it is used
     password: Password,
 }
 
+/// The type of managed state for the active guardian
 #[derive(Default)]
-struct ActiveAccountState(Mutex<Option<ActiveAccount>>);
+struct ActiveGuardianState(Mutex<Option<ActiveGuardian>>);
 
-/// Describes possible errors when importing an account.
-#[derive(thiserror::Error, Debug)]
-enum ImportWalletError {
-    /// An error happened while trying to write to disk
-    #[error("Could not save account: {0}")]
-    Write(#[from] std::io::Error),
-    /// Account has already been imported
-    #[error("Account already found in application")]
-    Duplicate,
+/// The collective state kept for all guardians
+#[derive(Default)]
+struct GuardiansState(Mutex<contract::GuardiansState>);
+
+/// The necessary election guard configuration to construct election guard
+/// entities.
+#[derive(Clone)]
+struct ElectionGuardConfig {
+    /// The election manifest
+    #[allow(dead_code)] // TODO: remove when it is used
+    manifest: ElectionManifest,
+    /// The election parameters
+    parameters: ElectionParameters,
 }
 
-impl serde::Serialize for ImportWalletError {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: serde::Serializer, {
-        serializer.serialize_str(self.to_string().as_ref())
+struct ElectionContractMarker;
+type ElectionClient = ContractClient<ElectionContractMarker>;
+
+#[derive(Clone)]
+struct ConnectionConfig {
+    node:     v2::Client,
+    contract: ElectionClient,
+}
+
+async fn get_resource_checked<J: DeserializeOwned>(url: &ChecksumUrl) -> Result<J, Error> {
+    let data = reqwest::get(url.url.clone()).await?.bytes().await?;
+
+    let hash = HashSha2256(sha2::Sha256::digest(&data).into());
+    if url.hash != hash {
+        return Err(anyhow!(
+            "Verification of remote resource at {} failed. Expected checksum {} did not match \
+             computed hash {}.",
+            url.url,
+            url.hash,
+            hash
+        )
+        .into());
     }
+
+    // It's fair to assume that data integrity check means that it can also be
+    // deserialized
+    let result = serde_json::from_slice(&data).unwrap();
+    Ok(result)
 }
 
+impl ConnectionConfig {
+    /// Creates a connection to a concordium node and a contract client. This
+    /// function panics if the necessary environment variables are not set.
+    async fn try_create_from_env() -> Result<Self, Error> {
+        #[allow(clippy::option_env_unwrap)] // To avoid cargo check errors in editor
+        let endpoint_var = option_env!("CCD_ELECTION_NODE")
```

Comment:
Who is expected to be setting these environment variables? Why are they "option_env" rather than just "env"?

Reply:
As explained in a reply to a previous comment, the only reason they are `option_env!` is because otherwise the editor shows an error if it cannot read the environment variable.

It is expected these are available at build time:

```bash
export CCD_ELECTION_CONTRACT_ADDRESS="<8011,0>"
export CCD_ELECTION_NODE="https://node.testnet.concordium.com:20000"
yarn tauri dev
```

I realise that this moves the error to runtime, so a workaround would be to ensure the environment variables can be read by `cargo check` in the editor context. What do you think?

Reply:
I commented on the other discussion with a proposed solution.